docs/choreolib/auto-factory.md

```diff
@@ -1,109 +1,282 @@
+- Conjoin a trigger to one created by `AutoRoutine` or `AutoTrajectory`
+
+```java
+Trigger myTrigger = new Trigger(() -> condition);
+
+// Safe
+routine.observe(myTrigger).onTrue(Commands.print("Foo"));
+routine.active().and(myTrigger).onTrue(Commands.print("Bar"));
+
+// Unsafe
+myTrigger.onTrue(Commands.print("Foo"));
+myTrigger.and(routine.active()).onTrue(Commands.print("Bar"));
+```
+
+Sometimes, you may want to implement a "branching auto": an autonomous routine that changes behavior based on the state of the robot. An excellent example for the need of branching autos is the [2024 season](https://youtu.be/9keeDyFxzY4), where robots would race to the midline to grab a gamepiece. If another robot beat yours to the midline, or your robot missed a game piece, a common strategy was to go directly to the next gamepiece on the midline, instead of coming back to score. As an example, [this match](https://youtu.be/_gcezRaGP5A?t=6) from the 2024 championship shows both 254 and 3339 running branching autos.
+
+Below is an example auto routine from the 2024 season, taking advantage of the `AutoRoutine` paradigm:
+
+!!! note
+    Be careful when using `Commands.either()` to achieve branching behavior. The resulting composition will require subsystems from both the `onTrue` and `onFalse` commands, which may block default commands that would otherwise be expected to run. As shown in the example below, chaining triggers is the recommended pattern for creating a branching auto.
+
+```java
+public AutoRoutine branching2024Auto() {
+    AutoRoutine routine = autoFactory.newRoutine("branching2024Auto");
+
+    // This routine uses segments between pre-defined handoff points.
+    // Expand the tooltip for information about their naming convention -> (1)
+
+    // Load the routine's trajectories
+    AutoTrajectory startToC2 = routine.trajectory("startToC2");
+    AutoTrajectory C2toM1 = routine.trajectory("C2toM1");
+    AutoTrajectory M1toM2 = routine.trajectory("M1toM2");
+    AutoTrajectory M2toM3 = routine.trajectory("M2toM3");
+    AutoTrajectory M1toScore = routine.trajectory("M1toScore");
+    AutoTrajectory M2toScore = routine.trajectory("M2toScore");
+    AutoTrajectory M3toScore = routine.trajectory("M3toScore");
+    AutoTrajectory scoreToM2 = routine.trajectory("scoreToM2");
+    AutoTrajectory scoreToM3 = routine.trajectory("scoreToM3");
+
+    // When the routine starts, reset odometry, shoot the first gamepiece, then go to the "C2" location
+    routine.active().onTrue(
+        Commands.sequence(
+            routine.resetOdometry(startToC2),
+            shooterSubsystem.shoot(),
+            startToC2.cmd()
+        )
+    );
+
+    // Pick up and shoot the gamepiece at the "C2" location, then go to the "M1" location
+    startToC2.active().whileTrue(intakeSubsystem.intake());
+    startToC2.done().onTrue(shooterSubsystem.shoot().andThen(C2toM1.cmd()));
+
+    // Run the intake when we are approaching a gamepiece
+    C2toM1.active()
```

Comment:
For stuff like this once #1002 is merged, doing a `routine.anyActive(scoreToM2, ...)` should be done to show off the functionality.